core/storage/blog/gae_models.py

```diff
@@ -496,3 +507,146 @@ def create(cls, blog_post_id: str, author_id: str) -> BlogPostRightsModel:
         entity.put()
 
         return entity
+
+
+class BlogAuthorDetailsModel(base_models.BaseModel):
+    """Model for storing user's blog author details.
+
+    The id/key of instances of this model is randomly generated string of
+    length 12.
+    """
+
+    # We use the model id as a key in the Takeout dict.
+    ID_IS_USED_AS_TAKEOUT_KEY: Literal[True] = True
+    # The user ID of the blog author.
+    author_id = datastore_services.StringProperty(indexed=True, required=True)
+    # The publicly viewable name of the user to display as author name in blog
+    # posts.
+    displayed_author_name = (
+        datastore_services.StringProperty(indexed=True, required=True))
+    # User specified biography to be shown on their blog author page.
+    author_bio = datastore_services.TextProperty(indexed=False)
+
+    @staticmethod
+    def get_model_association_to_user(
+    ) -> base_models.MODEL_ASSOCIATION_TO_USER:
+        """Model is exported as one instance per user."""
+        return base_models.MODEL_ASSOCIATION_TO_USER.ONE_INSTANCE_PER_USER
+
+    @staticmethod
+    def get_deletion_policy() -> base_models.DELETION_POLICY:
+        """Model contains data to pseudonymize corresponding to a user:
+        id field.
+        """
+        return base_models.DELETION_POLICY.LOCALLY_PSEUDONYMIZE
+
+    @classmethod
+    def has_reference_to_user_id(cls, user_id: str) -> bool:
+        """Check whether BlogAuthorUserModel exists for user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether any models refer to the given user ID.
+        """
+        return cls.query(
+            cls.author_id == user_id
+        ).get(keys_only=True) is not None
+
+    @classmethod
+    def export_data(cls, user_id: str) -> Dict[str, BlogAuthorDetailsModelDict]:
```

Comment:
It will result into the following error:
![image](https://user-images.githubusercontent.com/68547101/199314035-9c33d7a0-defa-4687-a737-3d06702165ca.png)

Reply:
But that needs to be solved differently, not by providing a wrong type.

You will need to add an ignore https://github.com/oppia/oppia/blob/develop/core/storage/user/gae_models.py#L3100-L3123 and probably create a new `TypedDict` that works for this function.